packages/strapi/lib/Strapi.js

```diff
@@ -414,35 +416,48 @@ class Strapi {
     return reload;
   }
 
-  async runBootstrapFunctions() {
-    const execBootstrap = async fn => {
+  async runLifecyclesFunctions(functionName) {
+    const execFunction = async fn => {
       if (!fn) return;
 
       return fn();
     };
 
-    // plugins bootstrap
+    const configPath = `functions.${functionName}`;
+
+    // plugins register
     const pluginBoostraps = Object.keys(this.plugins).map(plugin => {
-      return execBootstrap(_.get(this.plugins[plugin], 'config.functions.bootstrap')).catch(err => {
-        strapi.log.error(`Bootstrap function in plugin "${plugin}" failed`);
+      const pluginFunc = _.get(this.plugins[plugin], `config.${configPath}`);
+
+      return execFunction(pluginFunc).catch(err => {
+        strapi.log.error(`${functionName} function in plugin "${plugin}" failed`);
         strapi.log.error(err);
         strapi.stop();
       });
     });
+
     await Promise.all(pluginBoostraps);
 
-    // user bootstrap
-    await execBootstrap(_.get(this.config, ['functions', 'bootstrap']));
+    // user register
```

Comment:
Here and `L.444`, did you meant `user` & `admin` **function** instead of `register`?